shell/components/form/NameNsDescription.vue

```diff
@@ -220,39 +220,18 @@ export default {
       return this.nameDisabled || (this.mode === _EDIT && !this.nameEditable);
     },
 
-    namespaces() {
-      const currentStore = this.$store.getters['currentStore'](this.namespaceType);
-      const namespaces = this.namespacesOverride || this.$store.getters[`${ currentStore }/all`](this.namespaceType);
-      const filterNamespace = this.$store.getters['allNamespaces'];
-
-      const filtered = namespaces.filter( this.namespaceFilter || ((namespace) => {
-        // By default, include the namespace in the dropdown.
-        let out = true;
-
-        if (this.currentProduct?.customNamespaceFilter && this.currentProduct?.inStore) {
-          out = filterNamespace.find(NS => NS.metadata.name === namespace.metadata.name);
-        } else if (this.currentProduct?.hideSystemResources) {
-          // Filter out the namespace
-          // if it is a system namespace or if it is managed by
-          // Fleet.
-          out = !namespace.isSystem && !namespace.isFleetManaged;
-        }
-
-        if (this.mode === _CREATE) {
-          out = out && !!namespace.links.update;
```

Comment:
Not so sure if we are removing anything here.

Reply:
If a namespace is missing `links.update` the current user doesn't have permission to create resources within it, so it shouldn't be shown as an option in `NameNsDescription`. Read-only namespaces are still a valid choice in the `NamespaceFilter` since the user would be able to see resources within the namespace, so I think it makes sense to keep this logic within `NameNsDescription`

Reply:
I'm not sure why `isFleetManaged` is checked here but not in `NamespaceFilter` - I think this was added for Harvester specifically, maybe @WuJun2016 knows why we would be filtering fleet-managed namespaces from `NameNsDescription` and the namespace list view but not `NamespaceFilter`

Reply:
`NamespaceFilter` does not use the action, although we have to flag them as readonly in case we want to switch it to this as well. I'll have to think about this option, maybe by simply using the false value. 
The returned namespaces do not provide informations about the links, so or we extend it and change the logic or we set it in the state.

Reply:
What does `isFleetManaged` stand for?

Reply:
> 

@mantis-toboggan-md   This filtering rule was used by the old version of the `harvester`, although it is no longer in use in the `new version`. However, I believe it should still be retained now because we need to be compatible with `older` `harvester` versions, and we may remove it at some point in the future （such as rancher 2.8.0).

Reply:
@mantis-toboggan-md how can I create a namespace without `links.update`, so I can do a QA?

Reply:
Hi, @cnotv. You could follow the "Explorer Test case" to QA it. https://github.com/harvester/harvester/issues/1986#issuecomment-1261745712.